src/codegen/forward.jl

```diff
@@ -63,16 +64,13 @@ function fwd_transform!(ci, mi, nargs, N)
     end
 
     meth = mi.def::Method
-    nargs = Int(meth.nargs)
-    for i = 1:nargs
-        if meth.isva && i == nargs
-            args = map(i:(nargs+1)) do j::Int
-                emit!(Expr(:call, getfield, SlotNumber(2), j))
-            end
-            emit!(Expr(:(=), SlotNumber(2 + i), Expr(:call, ∂vararg{N}(), args...)))
-        else
-            emit!(Expr(:(=), SlotNumber(2 + i), Expr(:call, getfield, SlotNumber(2), i)))
-        end
+    nfixedargs = Int(meth.nargs) - meth.isva
+    for i in 1:nfixedargs
+        emit!(Expr(:(=), SlotNumber(2 + i), Expr(:call, getfield, SlotNumber(2), i)))
+    end
+    if meth.isva  # handle the varadic argument
+        vararg = emit!(Expr(:call, getfield, SlotNumber(2), nfixedargs+1))
+        emit!(Expr(:(=), SlotNumber(2 + i), Expr(:call, Core._apply_iterate, Base.iterate, ∂vararg{N}(), vararg)))
```

Comment:
idk if this is right, i don't know how to use `Core._apply_iterate`

Reply:
This isn't quite right, because you need the tail of the argument list rather than just the `nfixedargs+1`'st argument. So this either needs to rely on the `nargs` passed in, or do something like `SlotNumber(2)[(nfixedargs+1):end]`.